spec/factories/address.rb

```diff
@@ -49,9 +49,17 @@
         n
       end
 
-      sequence :country_iso do |n|
-        "country_iso_#{n}"
-      end
+      uprn { nil }
+      premises { nil }
+      street_address { nil }
+      locality { nil }
+      city { nil }
+      postcode { nil }
+      country_iso { nil }
```

Comment:
🤔 wondering what's the point to set up those as `nil`?

Reply:
The current address factory populates most of the address fields by default. If I didn't set the attributes to `nil` calls to `build(:address, :site)` would give me an object with the address type set to site, the grid reference and description populated, but also everything else filled in.

However this doesn't reflect what actually happens when you complete the site details in the journey. If you use a grid reference everything in the other address fields is set to nil.

So the expectation on `ConfirmationLetterPresenter.site_address_one_liner()` is that if the site has been set using a grid reference, it should return `""` because none of the address fields will have been set.

What I actually want to do is refactor the factories here to copy what you have done in the [WEX engine](https://github.com/DEFRA/waste-exemptions-engine/blob/master/spec/factories/address.rb) which would solve this problem. But I felt a wholesale refactoring of the address factory was beyond the scope of this PR hence these calls to `nil`.